tests/app/main/test_suppliers.py

```diff
@@ -1285,13 +1285,10 @@ def test_handles_api_errors_when_updating_supplier_or_contact_information(self):
         self.data_api_client.update_supplier.side_effect = [HTTPError(400, "I'm an error from the API"), None]
         self.data_api_client.update_contact_information.side_effect = HTTPError(400, "So am I!")
 
-        for error_message in ["I'm an error from the API", "So am I!"]:
+        for i in range(2):
             status, response = self.post_supplier_address_edit()
-            doc = html.fromstring(response)
 
-            assert status == 400
-            assert doc.xpath('normalize-space(//h1[@id="validation-masthead-heading"])') == error_message
-            assert "What is your registered office address?" in doc.xpath('//h1')[-1].text
+            assert status == 503
```

Comment:
I see what this test is doing but it's slightly confusing to read. I think it could benefit from either being separating out (less DRY) or parameterizing (maybe more complex at first glance). You're basically testing two things here and the setup for each is getting wrapped up with the other (e.g. if we remove the call to update_supplier from the code, the first iteration would still fail because it would silently catch the update_ci error).

Reply:
I see what you mean. I've tried it both ways, and separating into two tests is more clear I think. Parameterising it just added unnecessary complexity.